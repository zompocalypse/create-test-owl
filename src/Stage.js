import React from 'react';
import StageParticipants from './StageParticipants';
import store from './store';
import './Stage.css';

export default function Stage(props) {
  return (
    <section class="stage">
        <div class="active-view">
          {store.participants
          .filter(participant => participant.onStage === true)
          .map(participant =>
            <StageParticipants
              key={participant.id}
              name={participant.name}
              avatar={participant.avatar}
            />
          )}
        </div>
    </section>
  )
}