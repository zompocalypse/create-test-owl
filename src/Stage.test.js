import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './Stage';
import store from './store';

describe( 'Stage component' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Stage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<Stage
        {...store.participants}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})