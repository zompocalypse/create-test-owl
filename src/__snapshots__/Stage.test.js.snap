// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage component renders UI as expected 1`] = `
<section
  class="stage"
>
  <div
    class="active-view"
  >
    <div
      class="onStageParticipants"
    >
      <img
        alt="Koren Templeton"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <div
        class="stageDetails"
      >
        <span
          class="contact-on-stage-details-text"
        >
          Koren Templeton
        </span>
        <div
          class="mute"
        >
          mute
        </div>
        <div
          class="search"
        >
          search
        </div>
      </div>
    </div>
    <div
      class="onStageParticipants"
    >
      <img
        alt="Ashla Attwool"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <div
        class="stageDetails"
      >
        <span
          class="contact-on-stage-details-text"
        >
          Ashla Attwool
        </span>
        <div
          class="mute"
        >
          mute
        </div>
        <div
          class="search"
        >
          search
        </div>
      </div>
    </div>
  </div>
</section>
`;
